### PR TITLE
ARROW-11015: [CI][Gandiva] Move gandiva nightly build from travis to github action

### PR DESCRIPTION
**Pull request by projjal**

### CODE REVIEW

**Comment by projjal** [comment]
@github-actions crossbow submit -g gandiva

**Comment by projjal** [comment]
@kszucs Can you review this?

File: dev/tasks/gandiva-jars/github.linux.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          git -C arrow submodule update --init --recursive
+      - name: Build Gandiva
+        run: |
+          python3 -VV
+          cd arrow
+          mkdir -p dist
+          export CC="gcc-4.9" CXX="g++-4.9"
+          ulimit -c unlimited -S
+          docker run -v $PWD:/io -v $PWD:/arrow quay.io/projjal/arrow:gandivadocker /io/dev/tasks/gandiva-jars/build-cpp-linux.sh || exit 1
+          dev/tasks/gandiva-jars/build-java.sh || exit 1
```
Comment (kou) [inline dev/tasks/gandiva-jars/github.linux.yml:46]: How about using `set -e` instead of `|| exit 1`?

File: dev/tasks/gandiva-jars/github.osx.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        run: |
+          git clone --no-checkout {{ arrow.remote }} arrow
+          git -C arrow fetch -t {{ arrow.remote }} {{ arrow.branch }}
+          if [ $CROSSBOW_USE_COMMIT_ID = true ]; then git -C arrow checkout {{ arrow.head }}; else git -C arrow checkout FETCH_HEAD; fi
+          git -C arrow submodule update --init --recursive
+      - name: Build Gandiva
+        run: |
+          cd arrow
+          mkdir -p dist
+          export ARROW_TEST_DATA=$GITHUB_WORKSPACE/arrow/testing/data
```
Comment (kou) [inline dev/tasks/gandiva-jars/github.osx.yml:42]: How about using `$PWD` instead of `$GITHUB_WORKSPACE/arrow`?
```suggestion
          export ARROW_TEST_DATA=$PWD/testing/data
```

File: dev/tasks/gandiva-jars/github.linux.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          if [ $CROSSBOW_USE_COMMIT_ID = true ]; then git -C arrow checkout {{ arrow.head }}; else git -C arrow checkout FETCH_HEAD; fi
+          git -C arrow submodule update --init --recursive
+      - name: Build Gandiva
+        run: |
+          python3 -VV
+          cd arrow
+          mkdir -p dist
+          export CC="gcc-4.9" CXX="g++-4.9"
+          ulimit -c unlimited -S
+          docker run -v $PWD:/io -v $PWD:/arrow quay.io/projjal/arrow:gandivadocker /io/dev/tasks/gandiva-jars/build-cpp-linux.sh || exit 1
```
Comment (kou) [inline dev/tasks/gandiva-jars/github.linux.yml:45]: Can we simplify volume usage?

```suggestion
          docker run -v $PWD:/arrow quay.io/projjal/arrow:gandivadocker /arrow/dev/tasks/gandiva-jars/build-cpp-linux.sh || exit 1
```

File: dev/tasks/gandiva-jars/github.linux.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            jinja2 \
+            jira \
+            pygit2==1.0.3 \
+            ruamel.yaml \
+            setuptools_scm \
+            toolz
+      - name: Upload artifacts
+        run: |
+          python3 $GITHUB_WORKSPACE/arrow/dev/tasks/crossbow.py \
+            --queue-path $GITHUB_WORKSPACE/arrow \
```
Comment (kou) [inline dev/tasks/gandiva-jars/github.linux.yml:65]: Can we remove `$GITHUB_WORKSPACE/`?

```suggestion
          python3 arrow/dev/tasks/crossbow.py \
            --queue-path arrow \
```

File: dev/tasks/gandiva-jars/github.osx.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            jinja2 \
+            jira \
+            pygit2 \
+            ruamel.yaml \
+            setuptools_scm \
+            toolz
+      - name: Upload artifacts
+        run: |
+          python3 $GITHUB_WORKSPACE/arrow/dev/tasks/crossbow.py \
+            --queue-path $GITHUB_WORKSPACE/arrow \
```
Comment (kou) [inline dev/tasks/gandiva-jars/github.osx.yml:64]:
```suggestion
          python3 arrow/dev/tasks/crossbow.py \
            --queue-path arrow \
```

File: dev/tasks/tasks.yml [hunk 6]
```diff
@@ -43,7 +43,8 @@ groups:
     - centos-*-aarch64
 
   gandiva:
-    - gandiva-*
+    - gandiva-jar-ubuntu
+    - gandiva-jar-osx
```
Comment (kou) [inline dev/tasks/tasks.yml:47]: Could you revert this change?
`gandiva-*` is expanded to `gandiva-ja-ubuntu` and `gandiva-jar-osx` automatically.

**Comment by projjal** [comment]
@github-actions crossbow submit -g gandiva

**Comment by projjal** [comment]
@github-actions crossbow submit -g gandiva

**Review by kou — APPROVED** [review]
+1

**Comment by projjal** [comment]
@kou  Can you merge this PR. I don't have write access.

**Comment by kou** [comment]
Sure. Done.
